### DISCUSSION TITLE
Where/How to enable functionality?

### DISCUSSION THREAD

**Discussion by Leonardo-Ferreira** [OrchardCMS/OrchardCore#15970] [Q&A]
OrchardCore is actually huge. I didn't count but there's prob more than 100 projects so I think my question here is relevant.

What I would like to do, is, from a SaaS provider perspective, create a dashboard of my "services", where I would be able to know what my tenants are up to.

I would like to start by inspecting the databases under my management, and gather relevant informations such as sizes, availability, growth, load, index-fragmentation and others.

I thought about doing this using an abstraction layer and creating providers for each RDBMS (SQL Server, MySql, Azure Database, AWS RDS, others)

So first order of business: Where do I put the abstraction code? Is it a new project? if so, where? Is it just new code under a existing project? Which?

**Comment by sebastienros** [comment]
Looks like you want to create a custom module that would rely on multi-tenancy. You don't need to use the source code for that, or care about it, even though it might be easier to understand how everything works, and how the API is used.

In all cases, just create the projects that you need, like abstractions projects. And then a module project, that you would reference from your web app one.
